yapapi/strategy/wrapping_strategy.py

```diff
@@ -0,0 +1,29 @@
+from yapapi.props.builder import DemandDecorator, DemandBuilder
+
+from .base import MarketStrategy
+
+
+class WrappingMarketStrategy(DemandDecorator):
+    """
+    Helper class which allows other/user defined strategies to wrap some other strategies,
+    without overriding the attributes (e.g. defaults) defined on the derived-from strategy.
+
+    WrappingMarketStrategy classes are unusable on their own and always have to wrap some base
+    strategy.
+    """
+
+    base_strategy: MarketStrategy
+
+    def __init__(self, base_strategy: MarketStrategy):
+        """
+        :param base_strategy: the base strategy around which this strategy is wrapped
+        """
+        self.base_strategy = base_strategy
+
+    async def decorate_demand(self, demand: DemandBuilder):
```

Comment:
Why do we need a special case for `decorate_demand`?

Reply:
well, yes and no

theoretically no but given that `DemandDecorator` is abstract and requires implementation, practically, yes...